eng/native/init-vs-env.cmd

```diff
@@ -6,9 +6,8 @@
 set "__VCBuildArch="
 if /i "%~1" == "x86"   (set __VCBuildArch=x86)
 if /i "%~1" == "x64"   (set __VCBuildArch=x86_amd64)
```

Comment:
Is this line correct? Why are we using the x86 host bits when targeting x64?

Reply:
I wonder if `__VCBuildArch` should be initialized as `%PROCESSOR_ARCHITECTURE%_%~1` but with translating `x64->amd64` and `wasm->x64`. cc @jkotas

Reply:
I saw an older comment by @jkotas mentioning that the x64 toolset has a larger memory footprint without offering any tangible performance improvements over x86.

Reply:
I'm wondering if it might be an issue with our CI fleet which presumably is more resource constrained.

Reply:
My guess is that the CI machines are CPU constrained for C/C++ builds.